boa3_test/examples/HTLC.py

```diff
@@ -0,0 +1,234 @@
+ADDRESS_PREFIX: bytes = b'address'
+AMOUNT_PREFIX: bytes = b'amount'
+TOKEN_PREFIX: bytes = b'token'
+FUNDED_PREFIX: bytes = b'funded'
+
+# Number of seconds that need to pass before refunding the contract
+LOCK_TIME = 15 * 1
+
+NOT_INITIALIZED: bytes = b'not initialized'
+START_TIME: bytes = b'start time'
+SECRET_HASH: bytes = b'secret hash'
+DEPLOYED: bytes = b'deployed'
+
+
+# -------------------------------------------
+# Methods
+# -------------------------------------------
+
+
+@public
+def verify() -> bool:
+    """
+    When this contract address is included in the transaction signature,
+    this method will be triggered as a VerificationTrigger to verify that the signature is correct.
+    For example, this method needs to be called when withdrawing token from the contract.
+
+    :return: whether the transaction signature is correct
+    """
+    return check_witness(OWNER)
+
+
+@public
+def deploy() -> bool:
+    """
+    Initializes OWNER and change values of NOT_INITIALIZED and DEPLOYED when the smart contract is deployed.
+
+    :return: whether the deploy was successful. This method must return True only during the smart contract's deploy.
+    """
+    if not check_witness(OWNER):
+        return False
+    if get(DEPLOYED).to_bool() == True:
+        return False
+
+    put(OWNER, OWNER)
+    put(NOT_INITIALIZED, True)
+    put(DEPLOYED, True)
+    return True
+
+
+@public
+def atomic_swap(owner_address: UInt160, owner_token: bytes, owner_amount: int, other_person_address: UInt160,
+                other_person_token: bytes, other_person_amount: int, secret_hash: bytes) -> bool:
+    """
+    Initializes the storage when the atomic swap starts.
+
+    :param owner_address: address of owner
+    :type owner_address: UInt160
+    :param owner_token: other_person's desired token
+    :type owner_token: bytes
+    :param owner_amount: other_person's desired amount of tokens
+    :type owner_amount: int
+    :param other_person_address: address of other_person
+    :type other_person_address: bytes
+    :param other_person_token: owner's desired token
+    :type other_person_token: bytes
+    :param other_person_amount: owner's desired amount of tokens
+    :type other_person_amount: int
+    :param secret_hash: the secret hash created by the contract deployer
+    :type secret_hash: bytes
+
+    :return: whether the deploy was successful or not
+    :rtype: bool
+
+    :raise AssertionError: raised if `owner_address` or `other_person_address` length is not 20 or if `amount` is not
+    greater than zero.
+    """
+    # the parameters from and to should be 20-byte addresses. If not, this method should throw an exception.
+    assert len(owner_address) == 20 and len(other_person_address) == 20
+    # the parameter amount must be greater than 0. If not, this method should throw an exception.
+    assert owner_amount > 0 and other_person_amount > 0
+
+    if get(NOT_INITIALIZED).to_bool() == True and verify():
```

Comment:
same as before
is there any reason to not use `get(NOT_INITIALIZED).to_bool() and verify()`?